src/main/java/org/openstreetmap/atlas/geography/atlas/items/complex/roundabout/ComplexRoundabout.java

```diff
@@ -0,0 +1,295 @@
+        final double node1Y = edge1.start().getLocation().getLatitude().asDegrees();
+        final double node1X = edge1.start().getLocation().getLongitude().asDegrees();
+        final double node2Y = edge1.end().getLocation().getLatitude().asDegrees();
+        final double node2X = edge1.end().getLocation().getLongitude().asDegrees();
+        final double node3Y = edge2.end().getLocation().getLatitude().asDegrees();
+        final double node3X = edge2.end().getLocation().getLongitude().asDegrees();
+
+        // Get the vectors from node 2 to 1, and node 2 to 3
+        final double vector1X = node2X - node1X;
+        final double vector1Y = node2Y - node1Y;
+        final double vector2X = node2X - node3X;
+        final double vector2Y = node2Y - node3Y;
+
+        // The cross product tells us the direction of the orthogonal vector, which is
+        // Directly related to the direction of rotation/traffic
+        return (vector1X * vector2Y) - (vector1Y * vector2X);
+    }
+
+    public ComplexRoundabout(final Edge source)
+    {
+        this(source, Arrays.asList("AIA", "ATG", "AUS", "BGD", "BHS", "BMU", "BRB", "BRN", "BTN",
+                "BWA", "CCK", "COK", "CXR", "CYM", "CYP", "DMA", "FJI", "FLK", "GBR", "GGY", "GRD",
+                "GUY", "HKG", "IDN", "IMN", "IND", "IRL", "JAM", "JEY", "JPN", "KEN", "KIR", "KNA",
+                "LCA", "LKA", "LSO", "MAC", "MDV", "MLT", "MOZ", "MSR", "MUS", "MWI", "MYS", "NAM",
+                "NFK", "NIU", "NPL", "NRU", "NZL", "PAK", "PCN", "PNG", "SGP", "SGS", "SHN", "SLB",
+                "SUR", "SWZ", "SYC", "TCA", "THA", "TKL", "TLS", "TON", "TTO", "TUV", "TZA", "UGA",
+                "VCT", "VGB", "VIR", "WSM", "ZAF", "ZMB", "ZWE"));
+    }
+
+    public ComplexRoundabout(final Edge source, final List leftDrivingCountries)
+    {
+        super(source);
+        try
+        {
+            if (!(
+            // Make sure that the source has an iso_country_code
+            source.getTag(ISOCountryTag.KEY).isPresent()
+                    // Make sure that the source is an instances of a roundabout
+                    && JunctionTag.isRoundabout(source)
+                    // Make sure that we are looking at a master edge
+                    && (source.isMasterEdge())))
+            {
+                throw new CoreException("Invalid source edge for a roundabout.");
+            }
+            final String isoCountryCode = source.tag(ISOCountryTag.KEY).toUpperCase();
+
+            // Get all edges in the roundabout
+            this.roundaboutEdgeSet = new SimpleEdgeWalker(source, this.isRoundaboutEdge())
+                    .collectEdges();
+
+            // Try to build a Route from the edges
+            try
+            {
+                this.roundaboutRoute = Route.fromNonArrangedEdgeSet(this.roundaboutEdgeSet, false);
+            }
+            // If a Route cannot be formed, flag the edges as an incomplete roundabout.
```

Comment:
Since this isn't a check anymore, is "flag" still the right kind of language to be using?